### PR TITLE
[GEOS-10607] Links disappearing for the Admin user

### PR DESCRIPTION
**Pull request by ahmedababnehgeo**
[GEOS-10607](https://osgeo-org.atlassian.net/browse/GEOS-10607)

Whenever a user has been assigned to two roles, or even the same role, and one of those roles has ".admin" power on some data, while the other (or even the same) has ".write" power on the data, whenever you click on the "Layer Preview" link the others disappear.  

We add a new check to make sure that admin access is taken into account when retrieving Workspace Access limits.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [x] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Comment by taba90** [comment]
there is no test on this pr @ahmedababnehgeo. Please add one.

File: src/security/security-tests/src/test/java/org/geoserver/security/impl/DefaultResourceAccessManagerAuthTest.java [hunk 1]
```diff
@@ -38,6 +38,12 @@ public void testWideOpen() throws Exception {
         checkUserAccessFlat(manager, anonymous, true, true);
     }
 
+    @Test
+    public void testAdminProperties() throws Exception {
+        ResourceAccessManager manager = buildAccessManager("admin.properties");
+        checkUserAccessFlat(manager, root, true, true);
```
Comment (taba90) [inline src/security/security-tests/src/test/java/org/geoserver/security/impl/DefaultResourceAccessManagerAuthTest.java:44]: this method is testing that the canAccess methods return true according to the rules you set. I guess this was happening  even before your fix. The test should also test the getAccessLimits for the workspace doesn't return null but the WorkspaceLimits instance.

**Review by taba90 — CHANGES_REQUESTED** [review]

**Comment by ahmedababnehgeo** [comment]
@taba90 Test has been updated

**Comment by jodygarnett** [comment]
@taba90 has your feedback been addressed?

File: src/main/src/main/java/org/geoserver/security/impl/DefaultResourceAccessManager.java [hunk 2]
```diff
@@ -446,7 +446,7 @@ public WorkspaceAccessLimits getAccessLimits(Authentication user, WorkspaceInfo
 
         CatalogMode mode = getMode();
 
-        if (readable && writable) {
+        if (readable && writable && !adminable) {
```
Comment (aaime) [inline src/main/src/main/java/org/geoserver/security/impl/DefaultResourceAccessManager.java:449]: I'm confused by this change... looking at the ticket, we are dealing with a workspace administrator.
The code authorizing the components visibility should be the [WorkspaceAdminComponentAuthorizer](https://github.com/geoserver/geoserver/blob/6bb431a36a57aa1b6cf68d78bc7dea35b0958812/src/web/core/src/main/java/org/geoserver/web/WorkspaceAdminComponentAuthorizer.java#L52) which should allow visibility when there is at least one adminable workspace. Are you sure that, with this change, links are not getting visible also for users that cannot admin anything?

Comment (ahmedababnehgeo) [reply src/main/src/main/java/org/geoserver/security/impl/DefaultResourceAccessManager.java]: I tested the change for users who have no admin privilege at all, and links are not visible for them. It seems that this change solves the issue without side effects. 

Anyway, I changed the if statement organization, to increase its readability.

Comment (ahmedababnehgeo) [reply src/main/src/main/java/org/geoserver/security/impl/DefaultResourceAccessManager.java]: @aaime In order to make sure that REST API works well, I have tested access to the REST API page after this change. Normal users are denied access to REST API, whereas admins are allowed.

**Comment by aaime** [comment]
Tested manually, everything seems to work correctly. Merging.
